### PR TITLE
BI-8723 Add LP & LLP email mapping and logic

### PR DESCRIPTION
**Pull request by phudson-ch**
- Add company type specific BaseItemApi to CertificateOrderNotificationModel mapping logic 
- Add CertificateOptionsMapperFactory to provide company type specific CertificateOptionsMappers
- Add company type specific CertificateOptionsMappers for LP, LLP and Ltd [default] company types
- Added a ListHelper for mapping options to a list of strings
- Simplified directors and secretary mapper to use CertificateDetailsModelBuilder
- Added a new members options mapper, that also uses CertificateDetailsModelBuilder

Resolves: [BI-8723 (Confirmation e-mail)](https://companieshouse.atlassian.net/browse/BI-8723)

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateDetailsModelBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    private CertificateDetailsModelBuilder builder;
+
+    @BeforeEach
+    void setup() {
+        builder = new CertificateDetailsModelBuilder();
+    }
+
+    @Test
+    void correctlyBuildModelWhenListIsEmptyAndBasicInformationNull() {
+        CertificateDetailsModel expected = new CertificateDetailsModel(false, new ArrayList<String>() {
```
Comment (phudson-ch) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateDetailsModelBuilderTest.java:22]: false should be null

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateDetailsModelBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            {
+                add("No");
+            }
+        });
+        assertEquals(builder.build(), expected);
+    }
+
+    @Test
+    void correctlyBuildModelWhenListIsEmptyAndBasicInformationFalse() {
+        CertificateDetailsModel expected = new CertificateDetailsModel(false, new ArrayList<String>() {
```
Comment (phudson-ch) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateDetailsModelBuilderTest.java:34]: Add set basic information to false

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(MockitoExtension.class)
+public class CertificateOptionsMapperFactoryTest {
+
+    @Mock
+    private LLPCertificateOptionsMapper llpCertificateOptionsMapper;
+    @Mock
+    LPCertificateOptionsMapper lpCertificateOptionsMapper;
```
Comment (phudson-ch) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java]: Add private modifiers

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOrderNotificationMapperTest.java [hunk 4]
```diff
@@ -5,7 +5,7 @@
 import org.junit.jupiter.api.BeforeEach;
```
Comment (phudson-ch) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOrderNotificationMapperTest.java:4]: Check redundant code

**Review by phudson-ch — COMMENTED** [review]
Fix minor comments

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOrderNotificationMapperTest.java (hunk 4 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOrderNotificationMapperTest.java:4]: Redundant code removed

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java (hunk 3 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java]: Added private modifier

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateDetailsModelBuilderTest.java (hunk 2 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateDetailsModelBuilderTest.java:34]: Basic information has been set to false

**Review by kinpang-CH — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**